.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,17 @@
+on:
+  pull_request:
+    branches:
+      - '*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.0
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@08e2f20817b15149a52b5b3ebe7de50aff2ba8c5 # v3.4.0
```

Comment:
it's unclear to me how much golangci-lint-action does, but some things it's maybe missing:
- go build
- go test (-race?)
- gofmt
- go tidy

This is what we use in kube-openapi for example: https://github.com/kubernetes/kube-openapi/blob/master/.github/workflows/ci.yml

Reply:
definitely